pkg/cmd/cli/cmd/whoami.go

```diff
@@ -50,3 +54,39 @@ func (o WhoAmIOptions) WhoAmI() (*userapi.User, error) {
 
 	return me, err
 }
+
+func RunWhoAmI(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string, o *WhoAmIOptions) error {
+	if kcmdutil.GetFlagBool(cmd, "token") {
+		cfg, err := f.OpenShiftClientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		if len(cfg.BearerToken) == 0 {
+			return fmt.Errorf("no token is currently in use for this session")
+		}
+		fmt.Fprintf(out, "%s\n", cfg.BearerToken)
+		return nil
+	}
+	if kcmdutil.GetFlagBool(cmd, "context") {
```

Comment:
Whoami is fundamentally bound to the current context - so the default value (no flags) gets you the 90% case, then the 9% case is token (so you can try something out), then the last 1% is covered by context so you can go script something else.